eogrow/core/area/base.py

```diff
@@ -248,3 +249,92 @@ def _add_bbox_column(grid: List[gpd.GeoDataFrame]) -> None:
         for bbox_df in grid:
             crs = CRS(bbox_df.crs)
             bbox_df["BBOX"] = bbox_df.geometry.apply(lambda geo: BBox(geo.bounds, crs))  # noqa B023
+
+
+class BaseAreaManager(EOGrowObject, metaclass=ABCMeta):
+    """A manager for the AOI and how it is split into EOPatches"""
+
+    NAME_COLUMN = "eopatch_name"
+
+    class Schema(ManagerSchema):
+        ...
+
+    config: Schema
+
+    def __init__(self, config: Schema, storage: StorageManager):
+        """
+        :param config: A configuration file
+        :param storage: An instance of StorageManager class
+        """
+        super().__init__(config)
+
+        self.storage = storage
+
+    @abstractmethod
+    def get_area_geometry(self, *, crs: CRS = CRS.WGS84) -> Geometry:
+        """Provides a single geometry object of entire AOI"""
+
+    def get_grid(self) -> Dict[CRS, gpd.GeoDataFrame]:
+        """Provides a grid of bounding boxes which divide the AOI. Uses caching to avoid recalculations.
+
+        The grid is split into different CRS zones.
```

Comment:
```suggestion
        The grid is split into different UTM CRS zones.
```

Reply:
sadly, wishful thinking :)
i guess it's my fault for saying CRS zones instead of CRS parts/layers/whatever, but as we talked, we cannot afford to restrict ourselves to UTM zones only.

Reply:
Can you give an example how these parts are defined for other CRS cases? I thought this is only done for UTM CRS, because we have utility functions which detects the native UTM CRS of the EOPatch and it know in which part to put it.

Reply:
The responsibility of each manager is to provide a collection of geometries, each with it's respective CRS. From these geometries + CRS the BBoxes are then created.

In the case of `UtmZoneAreaManager` and `BatchAreaManager` the CRS will be UTM CRS. In the case of `CustomGridAreaManager` these will be whatever the geopackage wants. As discussed on the meeting we are trying to avoid streamlining everything to CRS. An example was the project done in epsg 3035 (if i recall correctly) which had reference data given in the mentioned CRS. Reprojecting to UTM CRS would produce warping artefacts as well as 'no-data' areas.